include/address.h

```diff
@@ -52,6 +52,11 @@ class SRAddress
     */
     SRAddress() : _is_tcp(true), _tcp_host(""), _tcp_port(0) {}
 
+    /*!
+    *   \brief SRAddress default destructor
+    */
+    virtual ~SRAddress() {}
```

Comment:
I think this might be the wrong solution.  This would be the ``SRAddress`` destructor doesn't do anything.   I think the error you were seeing is misleading and has nothing to do with the ``SRAddress`` class.  There is no dynamically allocated memory inside of this class that would require a custom destructor.

I think the real problem is in ``sharedmemorylist.tcc``.  I think the problem is that we allocate memory for a shared pointer using ``new[]`` but the shared pointer by default is using the ``delete`` operator to clean up memory when it should use ``delete[]``.  If we put a function like 

```c++
// Define memory deletion operator (delete[]) to
// match the memory allocation operator (new[])
class PointerDeletion {
    public:
        template<class T>
        void operator()(T* ptr) const {
            delete[] ptr;
        }
};
```

and then change line 36 of that file (https://github.com/CrayLabs/SmartRedis/blob/51c3dc6527df9ee9895678b4c534f9fe2b2c763c/include/sharedmemorylist.tcc#L36) to be:

```c++
    std::shared_ptr<T> s_ptr(ptr, PointerDeletion());
```
The warning will go away.  Do you think that is correct?

Reply:
I followed up on your suggestion. Inspecting the code, I tend to agree with you that you've identified a problem (albeit a pedantic one, but a real one nonetheless). I'll run the full suite tomorrow to confirm that the change holds